demos/custom-tabs-example-app/src/main/res/layout/activity_engagement_signals.xml

```diff
@@ -0,0 +1,103 @@
+        android:paddingRight="@dimen/activity_horizontal_margin"
+        android:paddingBottom="@dimen/activity_vertical_margin"
+        tools:context=".CustomUIActivity">
+
+        <LinearLayout
+            android:id="@+id/url_input"
+            android:layout_width="match_parent"
+            android:layout_height="48dp"
+            android:orientation="horizontal"
+            android:gravity="center_vertical"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent">
+
+            <TextView
+                android:id="@+id/url_label"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="@string/label_url"
+                />
+
+            <EditText
+                android:id="@+id/url"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:inputType="textUri"
+                android:text="@string/default_test_url"
+                />
+
+            <Button
+                android:id="@+id/start_custom_tab"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:padding="8dp"
+                android:text="@string/open"
+                />
+        </LinearLayout>
+
+        <TextView
+            android:id="@+id/label_event_navigation"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="16dp"
+            android:text="@string/label_event_navigation"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/url_input" />
+
+
+        <TextView
+            android:id="@+id/label_event_vertical_scroll"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="4dp"
+            android:text="@string/label_event_vertical_scroll"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/label_event_navigation" />
+
+        <TextView
+            android:id="@+id/label_event_greatest_percentage"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="4dp"
+            android:text="@string/label_event_greatest_percentage"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/label_event_vertical_scroll" />
+
+        <TextView
+            android:id="@+id/label_event_session_ended"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="4dp"
+            android:text="@string/label_event_session_ended"
```

Comment:
nit: It seems this and strings are always blank on `strings.xml`. IIRC, `android:text` is not required and you could just remove the `android:text` line and the empty strings from `strings.xml`